async/cohttp_async.ml

```diff
@@ -156,25 +156,60 @@ end
 
 module Client = struct
 
+  let read_request ic =
+    Response.read ic >>| function
+    | `Eof -> failwith "Connection closed by remote host"
+    | `Invalid reason -> failwith reason
```

Comment:
Yes that is clear, and seems to me to be the correct behavior. That allows one to process, and make decisions about retries.